src/modengine/ext/mod_loader/wwise_file_overrides.cpp

```diff
@@ -0,0 +1,120 @@
+// specifically sorts the WEMs into a subdir (`wem/`) and then another subdir 
+// based on the  first two digits of the WEM. So above example will spawn 
+// lookups in  `/wem/50/50846376.wem` and `enus/wem/50/50846376.wem`. In order 
+// to figure out if there is an override we will need to look in multiple 
+// directories per request. Luckily, aside from boot, this routine is called
+// quite infrequently.
+//
+// Also, worth pointing out that not all paths passed to this hook will have 
+// the `sd:/` prefix. So we cannot get away with the usual prefix / rewrite 
+// trick and will have to allocate a completely new string.
+
+namespace fs = std::filesystem;
+
+using namespace spdlog;
+
+std::optional<fs::path> find_override_file(const fs::path game_path)
+{
+    for (const auto& root : hooked_file_roots) {
+        trace(L"Searching for {} in {}", game_path.wstring(), root);
+
+        auto file_path = root / fs::path(game_path);
+        if (fs::exists(file_path)) {
+            return file_path;
+        }
+    }
+
+    return {};
+}
+
+const wchar_t* prefixes[3] = {
+    L"sd/",
+    L"sd/enus/",
+    L"sd/ja/",
+};
+
+std::optional<fs::path> check_paths(const std::wstring filename) {
+    for (auto prefix: prefixes) {
+        if (auto override = find_override_file(prefix+filename)) {
+            return override;
+        }
+    }
+
+    return {};
+}
+
+std::optional<fs::path> find_override(const std::wstring filename)
+{
+    // Check wem/<first to digits of filename>/<filename> too since ER uses 
+    // this format
+    if (filename.ends_with(L".wem")) {
+        auto wem_path = L"wem/" + filename.substr(0,2) + L"/" + filename;
+        auto wem_path_result = check_paths(wem_path);
+
+        if (wem_path_result.has_value()) {
+            return wem_path_result;
+        }
+    }
+
+    return check_paths(filename);
+}
+
+std::optional<std::wstring> normalize_filename(const std::wstring path) {
+    if (path.length() > 3 && path.substr(0, 4) == L"sd:/") {
+        return std::wstring(path.substr(4));
+    }
+
+    return {};
+}
+
+ScannedHook<decltype(&ak_file_location_resolver_open)> hooked_ak_file_location_resolver_open;
+
+void* __cdecl ak_file_location_resolver_open(UINT64 p1, wchar_t* path, UINT64 openMode, UINT64 p4, UINT64 p5, UINT64 p6)
+{
+    std::wstring lookup(path);
+    debug(L"sd.bhd entry requested: {}", lookup);
+
+    auto normalized = normalize_filename(lookup);
+    if (!normalized.has_value()) {
+        return hooked_ak_file_location_resolver_open.original(p1, path, openMode, p4, p5, p6);
+    }
+
+    auto filename = normalized.value();
+    auto override = find_override(filename);
+    if (!override.has_value()) {
+        return hooked_ak_file_location_resolver_open.original(p1, path, openMode, p4, p5, p6);
+    }
+
+    auto override_path_string = override.value().wstring();
+    return hooked_ak_file_location_resolver_open.original(p1, override_path_string.data(), 0, p4, p5, p6);
```

Comment:
Could we make an `AkOpenMode` enum reflecting the documentation up top just so it's obvious this `0` is `Read`?

Reply:
Should be resolved per fb2044adaddad599ffbb7217b95389419d24a2e7